module/zfs/dmu_objset.c

```diff
@@ -390,27 +391,12 @@ dmu_objset_byteswap(void *buf, size_t size)
 }
 
 /*
- * The hash is a CRC-based hash of the objset_t pointer and the object number.
+ * Runs cityhash on the objset_t pointer and the object number.
  */
 static uint64_t
 dnode_hash(const objset_t *os, uint64_t obj)
 {
-	uintptr_t osv = (uintptr_t)os;
-	uint64_t crc = -1ULL;
-
-	ASSERT(zfs_crc64_table[128] == ZFS_CRC64_POLY);
-	/*
-	 * The lower 11 bits of the pointer don't have much entropy, because
-	 * the objset_t is more than 1KB long and so likely aligned to 2KB.
-	 */
-	crc = (crc >> 8) ^ zfs_crc64_table[(crc ^ (osv >> 11)) & 0xFF];
-	crc = (crc >> 8) ^ zfs_crc64_table[(crc ^ (obj >> 0)) & 0xFF];
-	crc = (crc >> 8) ^ zfs_crc64_table[(crc ^ (obj >> 8)) & 0xFF];
-	crc = (crc >> 8) ^ zfs_crc64_table[(crc ^ (obj >> 16)) & 0xFF];
-
-	crc ^= (osv>>14) ^ (obj>>24);
-
-	return (crc);
+	return (cityhash2((uint64_t)os, obj));
```

Comment:
```suggestion
	return (cityhash2((uintptr_t)os, obj));
```
This is what done at `dbuf_hash()`. I worry compiler may not like conversion from 32bit pointer to 64bit integer.

Reply:
Even if we keep it `uintptr_t`, it would be cast to `uint64_t` when passing to `cityhash2`. And compilers will not issue any warning on this cast even with `-Wall -Wextra -pedantic`.

Reply:
There are architectures with 128bit pointers.  And I wonder if this code may produce there "cast from pointer to integer of different size".  Though I won't argue more about this, not my strongest area.  Leave to somebody else.  Just think coherency with `dbuf_hash()` would be good.